JitsiConference.js

```diff
@@ -1132,139 +1260,151 @@ JitsiConference.prototype.onIncomingCall
  * @param {String|null} reasonText human readable reason text which may provide
  * more details about why the call has been terminated.
  */
 JitsiConference.prototype.onCallEnded
-= function(JingleSession, reasonCondition, reasonText) {
-    logger.info(`Call ended: ${reasonCondition} - ${reasonText}`);
-    this.wasStopped = true;
+= function(jingleSession, reasonCondition, reasonText) {
+    logger.info(
+        `Call ended: ${reasonCondition} - ${reasonText
+            } P2P ?${jingleSession.isP2P}`);
+    if (jingleSession === this.jvbJingleSession) {
+        this.wasStopped = true;
+
+        // Send session.terminate event
+        Statistics.sendEventToAll('session.terminate');
+
+        // Stop the stats
+        if (this.statistics) {
+            this.statistics.stopRemoteStats();
+            this.statistics.stopCallStats();
+        }
 
-    // Send session.terminate event
-    Statistics.sendEventToAll('session.terminate');
+        // Current JVB JingleSession is no longer valid, so set it to null
+        this.jvbJingleSession = null;
 
-    // Stop the stats
-    if (this.statistics) {
-        this.statistics.stopRemoteStats();
-        this.statistics.stopCallStats();
+        // Let the RTC service do any cleanups
+        this.rtc.onCallEnded();
+    } else if (jingleSession === this.p2pJingleSession) {
+        this._stopP2PSession();
+    } else {
+        logger.error(
+            'Received onCallEnded for invalid session',
+            jingleSession.sid,
+            jingleSession.peerjid,
+            reasonCondition,
+            reasonText);
     }
-
-    // Current JingleSession is invalid so set it to null on the room
-    this.jingleSession = null;
-
-    // Let the RTC service do any cleanups
-    this.rtc.onCallEnded();
-
-    // PeerConnection has been closed which means that SSRCs stored in
-    // JitsiLocalTrack will not match those assigned by the old PeerConnection
-    // and SSRC replacement logic will not work as expected.
-    // We want to re-register 'ssrcHandler' of our local tracks, so that they
-    // will learn what their SSRC from the new PeerConnection which will be
-    // created on incoming call event.
-    const self = this;
-
-    this.getLocalTracks().forEach(localTrack => {
-        // Reset SSRC as it will no longer be valid
-        localTrack._setSSRC(null);
-
-        // Bind the handler to fetch new SSRC, it will un register itself once
-        // it reads the values
-        self.rtc.addListener(
-            RTCEvents.SENDRECV_STREAMS_CHANGED, localTrack.ssrcHandler);
-    });
 };
 
 /**
  * Handles the suspend detected event. Leaves the room and fires suspended.
+ * @param {JingleSessionPC} jingleSession
  */
-JitsiConference.prototype.onSuspendDetected = function() {
-    this.leave();
-    this.eventEmitter.emit(JitsiConferenceEvents.SUSPEND_DETECTED);
+JitsiConference.prototype.onSuspendDetected = function(jingleSession) {
```

Comment:
Yes, each JingleSessionPC instance emits events regardless of it's P2P status (the event is routed in JitsiConferenceEventManager).